### PR TITLE
BLADE-219 rootchain deploy command

### PR DESCRIPTION
**Pull request by dusannosovic-ethernal**
# Description

This PR deploy new bridge contracts

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

File: command/bridge/common/params.go [hunk 1]
```diff
@@ -206,34 +206,20 @@ func ExtractExitEventIDs(receipt *ethgo.Receipt) ([]*big.Int, error) {
 // ExtractChildTokenAddr extracts predicted deterministic child token address
 func ExtractChildTokenAddr(receipt *ethgo.Receipt, childChainMintable bool) (*types.Address, error) {
```
Comment (Stefan-Ethernal) [inline command/bridge/common/params.go]: Remove unused param.
```suggestion
func ExtractChildTokenAddr(receipt *ethgo.Receipt) (*types.Address, error) {
```

File: contracts/system_addresses.go [hunk 2]
```diff
@@ -39,6 +39,18 @@ var (
 	StakeManagerContract = types.StringToAddress("0x10022")
 	// StakeManagerContract is an address of stake manager contract on child chain
 	StakeManagerContractV1 = types.StringToAddress("0x100221")
+	// ValidatorSetStorageContract is an address of validator set storage contract on the child chain
```
Comment (Stefan-Ethernal) [inline contracts/system_addresses.go]:
```suggestion
	// ValidatorSetStorageContract is an address of validator set storage proxy contract
```

File: contracts/system_addresses.go [hunk 3]
```diff
@@ -39,6 +39,18 @@ var (
...
+	ValidatorSetStorageContract = types.StringToAddress("0x10031")
+	// ValidatorSetStorageContractV1 is an address of validator set storage contract on the child chain
+	ValidatorSetStorageContractV1 = types.StringToAddress("0x100311")
+	// GatewayCotnract is an address of gateway contract on child chain
```
Comment (Stefan-Ethernal) [inline contracts/system_addresses.go]:
```suggestion
	// GatewayCotnract is an address of gateway proxy contract
```

File: contracts/system_addresses.go [hunk 4]
```diff
@@ -39,6 +39,18 @@ var (
...
+	GatewayContract = types.StringToAddress("0x10032")
+	// GatewayCotnractV1 is an address of gateway contract on child chain
```
Comment (Stefan-Ethernal) [inline contracts/system_addresses.go]:
```suggestion
	// GatewayContractV1 is an address of gateway contract
```

File: contracts/system_addresses.go [hunk 5]
```diff
@@ -39,6 +39,18 @@ var (
...
+	GatewayContractV1 = types.StringToAddress("0x100321")
+	// BridgeStorageContract is an address of bridge storage contract on child chain
```
Comment (Stefan-Ethernal) [inline contracts/system_addresses.go]:
```suggestion
	// BridgeStorageContract is an address of bridge storage proxy contract
```

File: consensus/polybft/polybft.go [hunk 6]
```diff
@@ -276,7 +276,7 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
 						return err
 					}
 
-					if err = callContract(contracts.SystemCaller, contracts.RootMintableERC20PredicateContract, input,
+					if err = callContract(contracts.SystemCaller, contracts.RootERC20PredicateContract, input,
 						"RootMintableERC20PredicateAccessList", transition); err != nil {
```
Comment (Stefan-Ethernal) [inline consensus/polybft/polybft.go]:
```suggestion
						"RootERC20PredicateAccessList", transition); err != nil {
```

File: consensus/polybft/polybft.go [hunk 7]
```diff
@@ -288,8 +288,8 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
 						return err
 					}
 
-					if err = callContract(contracts.SystemCaller, contracts.RootMintableERC721PredicateContract, input,
-						"RootMintableERC721PredicateAccessList", transition); err != nil {
+					if err = callContract(contracts.SystemCaller, contracts.RootERC721PredicateContract, input,
+						"RootMintableERC20PredicateAccessList", transition); err != nil {
```
Comment (Stefan-Ethernal) [inline consensus/polybft/polybft.go]:
```suggestion
						"RootERC20PredicateAccessList", transition); err != nil {
```

File: consensus/polybft/polybft.go [hunk 8]
```diff
@@ -300,8 +300,8 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
 						return err
 					}
 
-					if err = callContract(contracts.SystemCaller, contracts.RootMintableERC1155PredicateContract, input,
-						"RootMintableERC1155PredicateAccessList", transition); err != nil {
+					if err = callContract(contracts.SystemCaller, contracts.RootERC1155PredicateContract, input,
+						"RootMintableERC20PredicateAccessList", transition); err != nil {
```
Comment (Stefan-Ethernal) [inline consensus/polybft/polybft.go]:
```suggestion
						"RootERC1155PredicateAccessList", transition); err != nil {
```

File: consensus/polybft/polybft.go [hunk 9]
```diff
@@ -358,8 +358,8 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
 						return err
 					}
 
-					if err = callContract(contracts.SystemCaller, contracts.RootMintableERC721PredicateContract, input,
-						"RootMintableERC721Predicate", transition); err != nil {
+					if err = callContract(contracts.SystemCaller, contracts.RootERC20PredicateContract, input,
```
Comment (Stefan-Ethernal) [inline consensus/polybft/polybft.go]: :exclamation_mark:
```suggestion
					if err = callContract(contracts.SystemCaller, contracts.RootERC721PredicateContract, input,
```

File: consensus/polybft/polybft.go [hunk 10]
```diff
@@ -358,8 +358,8 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
...
+						"RootERC20Predicate", transition); err != nil {
```
Comment (Stefan-Ethernal) [inline consensus/polybft/polybft.go]:
```suggestion
						"RootERC721Predicate", transition); err != nil {
```

File: consensus/polybft/polybft.go [hunk 11]
```diff
@@ -369,10 +369,11 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
 						return err
 					}
 
-					if err = callContract(contracts.SystemCaller, contracts.RootMintableERC1155PredicateContract, input,
-						"RootMintableERC1155Predicate", transition); err != nil {
+					if err = callContract(contracts.SystemCaller, contracts.RootERC20PredicateContract, input,
```
Comment (Stefan-Ethernal) [inline consensus/polybft/polybft.go]: :exclamation_mark:
```suggestion
					if err = callContract(contracts.SystemCaller, contracts.RootERC1155PredicateContract, input,
```

File: consensus/polybft/polybft.go [hunk 12]
```diff
@@ -369,10 +369,11 @@ func GenesisPostHookFactory(config *chain.Chain, engineName string) func(txn *st
...
+						"RootERC20Predicate", transition); err != nil {
```
Comment (Stefan-Ethernal) [inline consensus/polybft/polybft.go]:
```suggestion
						"RootERC1155Predicate", transition); err != nil {
```

File: command/bridge/deploy/deploy.go [hunk 13]
```diff
@@ -277,6 +279,46 @@ var (
...
+			config *polybft.RootchainConfig,
+			key crypto.Key,
+			chainID int64) error {
+			validatorSet, err := getValidatorSet(fmt, genesisValidators)
+			if err != nil {
+				return err
+			}
+
+			inputParams := &contractsapi.InitializeValidatorSetStorageFn{
+				NewBls:     config.BLSAddress,
```
Comment (Stefan-Ethernal) [inline command/bridge/deploy/deploy.go:290]:
```suggestion
			inputParams := &contractsapi.InitializeGatewayFn{
```

File: command/bridge/deploy/deploy.go [hunk 14]
```diff
@@ -277,6 +279,46 @@ var (
...
+				NewBn256G2: config.BN256G2Address,
+				Validators: validatorSet,
+			}
+
+			return initContract(fmt, relayer, inputParams, config.ValidatorSetStorage,
+				validatorSetStorageName, key)
+		},
```
Comment (Stefan-Ethernal) [inline command/bridge/deploy/deploy.go]: Do we need the `ValidatorSetStorage` deployed? I thought `Gateway` is enough.

**Review by oliverbundalo — APPROVED** [review]

File: command/bridge/deploy/deploy.go [hunk 15]
```diff
@@ -67,9 +66,6 @@ var (
 		getProxyNameForImpl(checkpointManagerName): func(rootchainConfig *polybft.RootchainConfig, addr types.Address) {
 			rootchainConfig.CheckpointManagerAddress = addr
 		},
-		getProxyNameForImpl(validatorSetStorageName): func(rootchainConfig *polybft.RootchainConfig, addr types.Address) {
-			rootchainConfig.ValidatorSetStorage = addr
```
Comment (Stefan-Ethernal) [inline command/bridge/deploy/deploy.go]: Remove `ValidatorSetStorage` field from the `RootchainConfig`?

**Pull request merged**